modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+
+.Error message
+[source, error]
+----
+Constraint already exists: Constraint( id=24, name='part_of', type='RELATIONSHIP PROPERTY TYPE', schema=()-[:PART_OF {order}]-(), propertyType=INTEGER )
+----
+
+======
+
+
+[[create-data-that-violates-a-constraint]]
+==== Creating data that violates existing constraints will fail
+
+This applies to all constraint types.
+
+.Create a node that violates an existing node property uniqueness constraint
+======
+
+.Create a `Book` node with an `isbn` that is already used in the graph
+[source, cypher, role=test-fail]
+----
+CREATE (book:Book {isbn: '1449356265', title: 'Graph Databases'})
+----
+
+In this case, the node is not created because the `isbn` property is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) already exists with label `Book` and property `isbn` = '1449356265'
+----
+
+======
+
+.Create a node that violates an existing node property existence constraint
+======
+
+.Create an `Author` node without a `name` property, given a property existence constraint on `:Author(name)`
+[source, cypher, role=test-fail]
+----
+CREATE (author:Author {surname: 'Austen'})
+----
+
+In this case, the node is not created because it is missing the `name` property which is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) with label `Author` must have the property `name`
+----
+
+======
+
+.Create a relationship that violates an existing relationship property type constraint
+======
+
+
+.Create a `PART_OF` relationship with a `STRING` `order` property, given a property type constraint on the relationship type `PART_OF` restricting the `order` property to `INTEGER` values
+[source, cypher, role=test-fail]
+----
+MATCH (movie:Movie {title:'Iron Man'}), (franchise:Franchise {name:'MCU'})
+CREATE (movie)-[part:PART_OF {order: '1'}]->(franchise)
+----
+
+In this case, the relationship is not created because the `order` property is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Relationship(0) with type `PART_OF` has property `order` of wrong type `String`. Allowed types: INTEGER
+----
+
+======
+
+
+.Create a node that violates existing node property key constraint
+======
+
+.Create an `Actor` node without a `firstname` property, given a node key constraint on `:Actor(firstname, surname)`
+[source, cypher, role=test-fail]
+----
+CREATE (actor:Actor {surname: 'Wood'})
+----
+
+In this case, the node is not created because it is missing the `firstname` property which is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) with label `Actor` must have the properties (`firstname`, `surname`)
+----
+
+======
+
+[[fail-to-create-constraint-due-to-existing-data]]
+==== Creating constraints when there exists conflicting data will fail
+
+This applies to all constraint types.
```

Comment:
same here as the section above, might be nice to actually give what data would violate depending on the constraint

maybe as a table? 🤔 
```
constraint type     | non-existing property | non-unique property/properties combination | property of wrong type
property uniqueness |                       |                      X                     |
property existence  |           X           |                                            |
property type       |                       |                                            |           X
key                 |           X           |                      X                     |
```

Reply:
feels especially important for this section as you don't even have an example for each constraint type